File: Programmers/[70129] 이진 변환 반복하기/KingDonggyu/70129.js
function solution(s) {
  const answer = [0, 0];

  while(s !== '1') {
    const c = s.replace(/0/g, '').length;
    answer[1] += s.length - c;
    s = c.toString(2);
    answer[0]++;
  }

  return answer;
}

File: Programmers/[70129] 이진 변환 반복하기/ONE/70129.java
class Solution {
    public int[] solution(String s) {
        int len;
        int[] answer = new int[2];

        while (!s.equals("1")) {
            len = s.codePoints()
                    .filter(c -> c == '1')
                    .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
                    .length();

            answer[0]++;
            answer[1] += s.length() - len;

            s = Integer.toBinaryString(len);
        }

        return answer;
    }
}